src/util.ts

```diff
@@ -238,3 +238,18 @@ export function logicalExpr<T>(op: LogicalOperand<T>, cb: Function): string {
 // Omit from http://ideasintosoftware.com/typescript-advanced-tricks/
 export type Diff<T extends string, U extends string> = ({[P in T]: P } & {[P in U]: never } & { [x: string]: never })[T];
 export type Omit<T, K extends keyof T> = {[P in Diff<keyof T, K>]: T[P]};
+
+// delete an object's empty key until it is not empty.
```

Comment:
/**
 * Delete nested property of an object, and delete the ancestors of the property if they become empty.
 */ 

Name it `deleteNestedProperty`.